apps/dav/lib/CalDAV/EventComparisonService.php

```diff
@@ -65,7 +65,7 @@ private function removeIfUnchanged(VEvent $filterEvent, array &$eventsToFilter):
 				$eventToFilterData[] = IMipService::readPropertyWithDefault($eventToFilter, $eventDiff, '');
 			}
 			// events are identical and can be removed
-			if (empty(array_diff($filterEventData, $eventToFilterData))) {
+			if ($filterEventData == $eventToFilterData) {
```

Comment:
do you have some example values for the two arrays?

Not sure if this breaks the logic but we always use strict comparison:

```suggestion
			if ($filterEventData === $eventToFilterData) {
```

Should be fine here, though, because both vars have the type array.

Reply:
Compared to `==`, the `===` operator also takes the order of array elements into consideration. Not familiar with those array contents and whether Nextcloud normalizes & sorts the data stored in those arrays. If not, I think that this could result in events treated as changed (with notifications being sent out) although the event content itself has not changed and confusing users. So `==` or `array_diff_assoc` would be a more defensive check.

Reply:
All three suggested solutions are equally valid here, as the function creates both arrays itself and in the same kind of controlled index sequence. My initial thought was as @flortsch pointed out that == would be more robust with the potential user data, but as @ChristophWurst  pointed out correctly, this is not necessary.

Eventually I also looked at the performance w.r.t. the array_diff_assoc solution:
`php -r '$st = microtime(true); $a1=array("","","",""); $a2=array("","","1",""); for($i = 1; $i <=100000; $i++) { if ($a1==$a2) {} else {} } echo "Execution time = ".(microtime(true)-$st)." sec\n";'
Execution time = 0.0091381072998047 sec`
`php -r '$st = microtime(true); $a1=array("","","",""); $a2=array("","","1",""); for($i = 1; $i <=100000; $i++) { if ($a1===$a2) {} else {} } echo "Execution time = ".(microtime(true)-$st)." sec\n";'
Execution time = 0.0064690113067627 sec`
`php -r '$st = microtime(true); $a1=array("","","",""); $a2=array("","","1",""); for($i = 1; $i <=100000; $i++) { if (empty(array_diff_assoc($a1,$a2))) {} else {} } echo "Execution time = ".(microtime(true)-$st)." sec\n";'
Execution time = 0.012899160385132 sec`

**And so the winner is ===**
[Well, actually performance measurement was as one would expect it to be, but still nice to be confirmed]

Reply:
Go for it :smile: thanks for the thorough analysis!